toolkit/tools/pkg/imagecustomizerlib/liveosisobuilder.go

```diff
@@ -324,28 +400,98 @@ func (b *LiveOSIsoBuilder) updateGrubCfg(savedConfigsFilePath string, grubCfgFil
 		return fmt.Errorf("failed to set SELinux mode:\n%w", err)
 	}
 
-	mergedConfigs, err := mergeConfigs(savedConfigsFilePath, extraCommandLine)
-	if err != nil {
-		return fmt.Errorf("failed to combine additional kernel command line parameters:\n%w", err)
-	}
-
 	liveosKernelArgs := fmt.Sprintf(kernelArgsLiveOSTemplate, liveOSDir, liveOSImage)
-	additionalKernelCommandline := liveosKernelArgs + " " + string(mergedConfigs.Iso.KernelCommandLine.ExtraCommandLine)
+	additionalKernelCommandline := liveosKernelArgs + " " + string(savedConfigs.Iso.KernelCommandLine.ExtraCommandLine)
 
 	inputContentString, err = appendKernelCommandLineArgsAll(inputContentString, additionalKernelCommandline,
 		true /*allowMultiple*/, false /*requireKernelOpts*/)
 	if err != nil {
 		return fmt.Errorf("failed to update the kernel arguments with the LiveOS configuration and user configuration in the iso grub.cfg:\n%w", err)
 	}
 
-	err = mergedConfigs.persistSavedConfigs(savedConfigsFilePath)
+	err = file.Write(inputContentString, isoGrubCfgFileName)
 	if err != nil {
-		return fmt.Errorf("failed to save iso configs:\n%w", err)
+		return fmt.Errorf("failed to write %s:\n%w", isoGrubCfgFileName, err)
 	}
 
-	err = file.Write(inputContentString, grubCfgFileName)
+	err = verifyDracutPXESupport(savedConfigs.OS.DracutPackageInfo)
 	if err != nil {
-		return fmt.Errorf("failed to write %s:\n%w", grubCfgFileName, err)
+		logger.Log.Warnf("cannot generate grub.cfg for PXE booting.\n%v", err)
```

Comment:
Why are you only warning here?

Reply:
The user does not ask explicitly to create a PXE compatible image (with grub-pxe.cfg). The user may just want an ISO.
So, if dracut version is below the PXE required version, we shouldn't fail the ISO generation.
This is true for Mariner 2.0, for example.

Reply:
I'd like to reserve WARN for when something bad happened (e.g. cleanup failed). This should probably be downgraded to INFO or DEBUG.

Also, please add a comment explaining why it is ok if this fails. Perhaps something like:

```golang
// As a convenience for the user, for images that can support PXE booting, create a version of the grub.cfg file that can be used for PXE booting and add it to the ISO.
```

Reply:
I've switched it to an `info` - however, I think it should be a warning.
The API is such that we generate a PXE bootable liveos iso by default. If we can't hold that promise, I think we should want the user and let them decide. It's 'bad' if the user wants to do a PXE deployment.

Reply:
INFO is correct. WARN should be limited to things that likely require user attention. If it can ever be expected behavior, then it doesn't get to be WARN.